base/src/test/resources/success/common/src/Arith/Nat/Properties.aya

```diff
@@ -10,3 +16,42 @@ def z≠s {a : Nat} (p : 0 = suc a) : False => (\i => diag (p i)).coe a
 private def diag Nat : Type
 | 0 => Nat
 | suc a => False
+
+private def suc-inj (m : Nat) : Nat
+| 0 => 114514
+| suc m => m
+def s=s {m n : Nat} (p : suc m = suc n) : m = n => (\ i => suc-inj (p i))
+
+def =?-refl (m : Nat) : (m =? m) = true
+| 0 => idp
+| suc m' =>
+  // Goal : (suc m' =? suc m?) = true
+  //      : (m' =? m') = true
+  =?-refl m'
+
+def =?-= {m n : Nat} (p : (m =? n) = true) : m = n
+| {0}, {0}, _ => idp
+| {0}, {suc n'}, p => elim (t≠f (sym p))
+| {suc m'}, {0}, p => elim (t≠f (sym p))
+| {suc m'}, {suc n'}, p =>
+  // p : (suc m' =? suc n') = true
+  //   : (m' =? n') = true            simpl.
+  // Goal : suc m' = suc n'
+  pmap suc (=?-= {m'} {n'} p)
```

Comment:
Is it possible to do it w/o these implicits?

Reply:
🤔what is `w/o`?

Reply:
It's 'without'

Reply:
These comments can be deleted. But it helps someone who reads this code.